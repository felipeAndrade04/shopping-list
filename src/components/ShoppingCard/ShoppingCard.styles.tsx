import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: 0.5,
}))`
  padding: 12px;
  background: ${({ theme }) => theme.colors.white};
  border-radius: 8px;
  box-shadow: 4px 4px 2px #0d0a0b40;
`;

export const Title = styled.Text`
  font-size: 16px;
  font-family: ${({ theme }) => theme.fonts.bold};
  color: ${({ theme }) => theme.colors.dark};
`;

export const Footer = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const FooterLeft = styled.View`
  flex-direction: row;
  align-items: center;
`;

export const Counter = styled.Text`
  font-size: 16px;
  color: ${({ theme }) => theme.colors.gray[300]};
  font-family: ${({ theme }) => theme.fonts.medium};
`;

export const CounterTotal = styled.Text`
  font-size: 16px;
  color: ${({ theme }) => theme.colors.dark};
  font-family: ${({ theme }) => theme.fonts.bold};
`;

export const Date = styled.Text`
  font-size: 14px;
  color: ${({ theme }) => theme.colors.gray[300]};
  font-family: ${({ theme }) => theme.fonts.regular};
`;
