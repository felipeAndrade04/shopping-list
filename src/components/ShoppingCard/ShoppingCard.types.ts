import { Shopping } from '@app/models';

export interface ShoppingCardProps {
  shopping: Shopping;
}
