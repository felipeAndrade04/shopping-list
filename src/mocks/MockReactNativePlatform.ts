jest.mock('react-native/Libraries/Utilities/Platform', () => {
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  const fs = require('fs');
  const json = JSON.parse(fs.readFileSync('node_modules/react-native/package.json', 'utf8'));
  const platform = jest.requireActual('react-native/Libraries/Utilities/Platform');
  const version = json.version.toString().split('.');
  return {
    ...platform,
    constants: {
      ...platform.constants,
      reactNativeVersion: {
        major: version[0],
        minor: version[1],
        patch: version[2],
      },
    },
  };
});
