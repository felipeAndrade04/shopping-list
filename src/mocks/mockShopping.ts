import { Shopping } from '@app/models';

export const mockedShopping: Shopping = {
  products: [],
  name: 'Lista 8',
  created_at: new Date('01/01/2023'),
  id: '123',
};
