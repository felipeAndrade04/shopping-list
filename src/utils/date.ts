const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
