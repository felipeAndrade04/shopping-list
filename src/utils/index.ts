export * from './delay';
export * from './date';
