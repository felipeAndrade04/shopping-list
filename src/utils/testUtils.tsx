import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';
import { MockSafeAreaContext } from '../mocks';
import { NavigationContainer } from '@react-navigation/native';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>
      <NavigationContainer>
        <MockSafeAreaContext.SafeAreaProvider>{children}</MockSafeAreaContext.SafeAreaProvider>
      </NavigationContainer>
    </Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
