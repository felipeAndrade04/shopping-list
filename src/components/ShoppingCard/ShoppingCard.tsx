import React from 'react';
import * as S from './ShoppingCard.styles';
import { ShoppingCardProps } from './ShoppingCard.types';
import { Spacer } from '../Spacer';
import { formatDate } from '@app/utils';
import { ProgressCircle } from '../ProgressCircle';
import { useNavigation } from '@react-navigation/native';
import { ShoppingListStackNavigationProps } from '@app/navigation/stackNavigation/shoppingList';
import { Product } from '@app/models';

export function ShoppingCard({ shopping }: ShoppingCardProps) {
  const navigation = useNavigation<ShoppingListStackNavigationProps>();

  function calcProgress(products: Product[]) {
    if (products.length === 0) {
      return 0;
    }

    return 100;
  }

  function navigateToShopping() {
    navigation.navigate('ShoppingList', { id: shopping.id });
  }

  return (
    <S.Container onPress={navigateToShopping}>
      <S.Title>{shopping.name}</S.Title>

      <Spacer dimesion={8} />

      <S.Footer>
        <S.FooterLeft>
          <ProgressCircle progress={calcProgress(shopping.products)} size={25} />
          <Spacer dimesion={8} />
          <S.Counter>
            <S.CounterTotal>{shopping.products.length}</S.CounterTotal>/{shopping.products.length}
          </S.Counter>
        </S.FooterLeft>
        <S.Date>{formatDate(shopping.created_at)}</S.Date>
      </S.Footer>
    </S.Container>
  );
}
