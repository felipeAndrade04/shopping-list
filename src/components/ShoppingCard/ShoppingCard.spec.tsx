import React from 'react';
import { render } from '@testing-library/react-native';
import { wrapper } from '@app/utils/testUtils';
import { ShoppingCard } from './ShoppingCard';
import { mockedShopping } from '@app/mocks/mockShopping';

describe('<ShoppingCard />', () => {
  it('Should render correctly', () => {
    const tree = render(<ShoppingCard shopping={mockedShopping} />, { wrapper }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
