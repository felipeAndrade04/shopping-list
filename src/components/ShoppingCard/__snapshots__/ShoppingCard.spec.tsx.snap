// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingCard /> Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "shadowColor": "#0d0a0b40",
      "shadowOffset": Object {
        "height": 4,
        "width": 4,
      },
      "shadowOpacity": 1,
      "shadowRadius": 2,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#0D0A0B",
          "fontFamily": "Poppins-Bold",
          "fontSize": 16,
        },
      ]
    }
  >
    Lista 8
  </Text>
  <View
    dimesion={8}
    style={
      Array [
        Object {
          "height": 8,
          "width": 8,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        collapsable={false}
        size={25}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#ACACAC",
            "borderRadius": 12.5,
            "borderWidth": 4.166666666666667,
            "height": 25,
            "justifyContent": "center",
            "transform": Array [
              Object {
                "rotate": "-45deg",
              },
            ],
            "width": 25,
          }
        }
      >
        <View
          collapsable={false}
          size={25}
          style={
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#0D0A0B",
              "borderRadius": 12.5,
              "borderRightColor": "transparent",
              "borderTopColor": "#0D0A0B",
              "borderWidth": 4.166666666666667,
              "height": 25,
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 25,
            }
          }
        />
        <View
          collapsable={false}
          size={25}
          style={
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#ACACAC",
              "borderRadius": 12.5,
              "borderRightColor": "transparent",
              "borderTopColor": "#ACACAC",
              "borderWidth": 4.166666666666667,
              "height": 25,
              "position": "absolute",
              "width": 25,
            }
          }
        />
        <View
          collapsable={false}
          size={25}
          style={
            Object {
              "borderBottomColor": "transparent",
              "borderLeftColor": "#0D0A0B",
              "borderRadius": 12.5,
              "borderRightColor": "transparent",
              "borderTopColor": "#0D0A0B",
              "borderWidth": 4.166666666666667,
              "height": 25,
              "opacity": 0,
              "position": "absolute",
              "transform": Array [
                Object {
                  "rotate": "0deg",
                },
              ],
              "width": 25,
            }
          }
        />
      </View>
      <View
        dimesion={8}
        style={
          Array [
            Object {
              "height": 8,
              "width": 8,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ACACAC",
              "fontFamily": "Poppins-Medium",
              "fontSize": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#0D0A0B",
                "fontFamily": "Poppins-Bold",
                "fontSize": 16,
              },
            ]
          }
        >
          0
        </Text>
        /
        0
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#ACACAC",
            "fontSize": 14,
          },
        ]
      }
    >
      0 de jan
    </Text>
  </View>
</View>
`;
